backend/mysql_utils.js

```diff
@@ -0,0 +1,228 @@
+                }
+            });
+        } else {
+            return dbError(err, callback);
+        }
+    });
+}
+
+exports.loginUser = function (email, pass, mysqlConnPool, callback) {
+    const sql = `SELECT * FROM users WHERE email_id=? and password=?`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, [email, pass], function (err, result) {
+                conn.release();
+                if (!err && result.length === 1) {
+                    return callback(null, result);
+                } else {
+                    return dbError(err, callback);
+                }
+            });
+        } else {
+            return dbError(err, callback);
+        }
+    });
+}
+
+exports.getUserConfig = function (bucketId, mysqlConnPool, callback) {
+    const sql = `SELECT value FROM buckets WHERE bucket_id = ${bucketId}`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, result) {
+                conn.release();
+                if (err) {
+                    console.error(`error getting user configurations: ${err.toString()}`);
+                    return dbError(err, callback);
+                } else {
+                    const grants = [];
+                    if (result.length === 1) {
+                        try {
+                            const grant = JSON.parse(result[0].value);
+                            grant.type = "s3";
+                            grants.push(grant);
+                        } catch (ex) {
+                            console.error(`error parsing user configurations: ${ex.toString()}`);
+                            return dbError(ex, callback);
+                        }
+
+                    }
+                    return callback(null, { grants });
+                }
+            });
+        } else {
+            return dbError(err, callback);
+        }
+    });
+}
+
+
+exports.storeRecord = function (userId, recordName, baseUrl, recordId, mysqlConnPool) {
+    const sql = `INSERT INTO records(id, name, creator, link) VALUES ('${recordId}', '${recordName}','${userId}', '${baseUrl}')`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, result) {
+                conn.release();
+                if (err) {
+                    console.error("error storing record link")
+                }
+            });
+        }
+    });
+}
+
+exports.deleteRecord = function (recordId, mysqlConnPool) {
+    const sql = `DELETE FROM records WHERE id='${recordId}'`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, result) {
+                conn.release();
+                if (err) {
+                    console.error(`error deleting record link: ${err}`)
+                }
+            });
+        }
```

Comment:
Shouldn't we handle `else` case as well?

Reply:
done